scipy/stats/_entropy.py

```diff
@@ -403,9 +404,9 @@ def _ebrahimi_entropy(X, m, *, xp):
     differences = X[..., 2 * m:] - X[..., : -2 * m:]
 
     i = xp.arange(1, n+1, dtype=X.dtype)
-    ci = xp.ones_like(i)*2
-    ci[i <= m] = 1 + (i[i <= m] - 1)/m
-    ci[i >= n - m + 1] = 1 + (n - i[i >= n-m+1])/m
+    ci = xp.where(i <= m, 1 + (i - 1)/m, xp.ones_like(i)*2)
```

Comment:
I suppose this could be `xp.asarray(2., dtype=X.dtype)`, but let's leave it alone until 2024.12 of the standard comes out. (Then it can just be, you know, `2.`).

Reply:
2024.12 is out now for numpy, cupy, array-api-strict, dask, and torch.
JAX is still at 2023.12 but it may already support this bit.